testkit/src/mill/testkit/IntegrationTester.scala

```diff
@@ -91,6 +96,49 @@ object IntegrationTester {
       )
     }
 
+    private val evalAsyncCounter = new AtomicInteger
+    def evalAsync(
+        cmd: os.Shellable,
+        env: Map[String, String] = millTestSuiteEnv,
+        cwd: os.Path = workspacePath,
+        stdin: os.ProcessInput = os.Pipe,
+        stdout: os.ProcessOutput = os.Pipe,
+        stderr: os.ProcessOutput = os.Pipe,
+        mergeErrIntoOut: Boolean = false,
+        timeout: Long = -1,
+        check: Boolean = false,
+        propagateEnv: Boolean = true,
+        timeoutGracePeriod: Long = 100
+    ): Future[IntegrationTester.EvalResult] = {
```

Comment:
Rather than duplicating `eval` and having to duplicate the entire method signature and maintain it going forward, can we instead spawn a `ThreadPoolExecutor` and use `Future{eval()}` calls? The performance difference is immaterial since these are small test cases, and it would simplify the code considerably

Reply:
👍  I was hesitating to do something along those lines